healthx/handler.go

```diff
@@ -64,6 +62,10 @@ type Handler struct {
 	ReadyChecks   ReadyCheckers
 }
 
+type Options struct {
+	Middleware func(http.Handler) http.Handler
+}
```

Comment:
Why is this a struct instead of a function?

```go
type Options func(*options)
```

Wouldn't it then be easier to add options through functions? With this method we expose the `Middleware` struct field to the outside - which is probably not what we want?

e.g.

```go
WIthMiddleware(m http.Handler) Options {
  return func(o *options) {
    o.middleware = m
  }
}

Reply:
There isn't a `options` struct in the healthx, I'd still have to write a struct for `options` , just did it on the original struct.

As for exposing the type, need a confirmation.
`Options` being exposed is fine (cause well you need the reference in the used functions), and `Middleware` isn't, right? Will be changed. 

Thought, I might need an explanation as to why exposing it would be a bad decision, if you don't mind that is